sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/rx/OrderbyDocumentQueryTest.java

```diff
@@ -9,6 +9,13 @@
 import com.azure.cosmos.CosmosBridgeInternal;
 import com.azure.cosmos.CosmosClientBuilder;
 import com.azure.cosmos.CosmosException;
+import com.azure.cosmos.implementation.AsyncDocumentClient;
+import com.azure.cosmos.implementation.PartitionKeyRange;
+import com.azure.cosmos.implementation.apachecommons.lang.tuple.Pair;
+import com.azure.cosmos.implementation.query.ItemComparator;
```

Comment:
your test validation relies on `ItemComparator` which essentially means the validation uses the existing implementation comparator to validate the expected order.

This is not ideal as if I go and change ItemComparator in the implementation and introduce a bug or change the comparer and its ordering, still this test will pass.

For the validation you shouldn't have dependency on the implementation.

Please try to find the expected order differently without relying on the implementation.

please remove `import com.azure.cosmos.implementation.query.ItemComparator;` import we should not have any dependency on the implementation for validating ordering otherwise the test will not capture bugs we add in the implementation.

Reply:
Removed ItemComparator and also improved the test to check for proper grouping of results and sorting inside each group